lib/generators.nix

```diff
@@ -242,30 +242,21 @@ rec {
     */
   toYAML = {}@args: toJSON args;
 
-  withRecursion =
-    args@{
-      /* If this option is not null, the given value will stop evaluating at a certain depth */
-      depthLimit
-      /* If this option is true, an error will be thrown, if a certain given depth is exceeded */
-    , throwOnDepthLimit ? true
-    }:
-      assert builtins.isInt depthLimit;
-      let
-        transform = depth:
-          if depthLimit != null && depth > depthLimit then
-            if throwOnDepthLimit
-              then throw "Exceeded maximum eval-depth limit of ${toString depthLimit} while trying to evaluate with `generators.withRecursion'!"
-              else const "<unevaluated>"
-          else id;
-        mapAny = with builtins; depth: v:
-          let
-            evalNext = x: mapAny (depth + 1) (transform (depth + 1) x);
-          in
-            if isAttrs v then mapAttrs (const evalNext) v
-            else if isList v then map evalNext v
-            else transform (depth + 1) v;
-      in
-        mapAny 0;
+  withRecursionLimit = maxDepth: fn:
+    let
+      wrapper = { value, depth }: fn {
```

Comment:
This is a breaking API change of an exposed library function, we can’t do that.

Reply:
You need to keep the API the same; e.g. making some of the options no-ops should also work.

Reply:
That's not really viable here unfortunately - the old version took an object to be inspected, and the new one takes a function to be wrapped. I guess we can make a `withRecursionLimit'`, or even a `_withRecursionLimit'`?

Reply:
As I said, I don’t think we should break the API without good cause.

Reply:
Yes, what I'm saying is, keep the old function as it is, introduce the new one without exposing it.